lib/velum/file_config.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+require "json"
+require "fileutils"
+require "securerandom"
+
+module Velum
+  # Handle IO operations and data serialisation for configuration files under a specific directory.
+  class FileConfig
+    def initialize(base_dir)
+      @base_dir = base_dir
+    end
+
+    # Return true only if the specified file name exists in the configuration base directory.
+    def exists?(file_name)
+      File.exists?(File.join(@base_dir, file_name))
+    end
+
+    # Deserialise JSON text from the specified file into a hash structure and return.
+    def read_hash(file_name)
+      JSON.parse(IO.read(File.join(@base_dir, file_name)))
+    end
+
+    # Serialise the hash structure into JSON text and overwrite the specified file with it.
+    def write_hash(file_name, hash)
+      FileUtils::mkdir_p(@base_dir)
+      IO.write(File.join(@base_dir, file_name), JSON.dump(hash))
+    end
+  end
+
+  # Manage persistent application-wide secret data.
+  class AppSecrets
```

Comment:
as the app secrets are in the center of that feature why not call the library sth like `app_secrets.rb`? `file_config.rb` doesn't say much